### PR TITLE
Ct 3910 Add partial case flag for closed Offender SAR case based on new design

### PR DESCRIPTION
**Pull request by ymao2**
## Description

This PR is to add partial case feature for closed Offender SAR case
## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [X] (3) Tests passing
* [X] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
* [X] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [X] (8) Data migration script is created if any of letter templates is changed

### Screenshots

1. case detail page 
Screenshot 2021-11-11 at 22 13 26

Screenshot 2021-11-11 at 22 12 30

2. edit page

Screenshot 2021-11-11 at 22 12 55
Screenshot 2021-11-11 at 22 12 48

3. case history
Screenshot 2021-11-11 at 22 13 50

### CODE REVIEW

File: app/views/cases/offender_sar/_partial_case_flags.html.slim [hunk 1]
```diff
@@ -0,0 +1,28 @@
+.grid-row
+  .column-full.data-requests
+    h2.request--heading.data-requests__title
```
Comment (digitalali-moj) [inline app/views/cases/offender_sar/_partial_case_flags.html.slim:3]: Can we remove the data-requests__title class?

File: app/views/cases/offender_sar/_partial_case_flags.html.slim [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    - if !case_details.is_partial_case.nil?
+
+      table#data-request-table.report
+        thead
+          tr
+            th Confirmed SSCL COVID-19 partial case
+            th Date letter sent
+            th SSCL managing case
+            th
```
Comment (digitalali-moj) [inline app/views/cases/offender_sar/_partial_case_flags.html.slim:14]: Do we need the extra th?

File: app/models/case/sar/offender.rb [hunk 3]
```diff
@@ -88,6 +92,12 @@ def searchable_fields_and_ranks
     third_party_recipient: 'third_party_recipient',
   }
 
+  enum further_actions_required: {
```
Comment (digitalali-moj) [inline app/models/case/sar/offender.rb:95]: Can these be translated (if it's not a pain!)

File: app/views/cases/offender_sar/_partial_case_flags.html.slim [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        tbody
+          tr
+            td
+              = case_details.is_partial_case? ? "Yes" : "No"
```
Comment (digitalali-moj) [inline app/views/cases/offender_sar/_partial_case_flags.html.slim:18]: Can these be translated?

File: app/views/cases/offender_sar/_partial_case_flags.html.slim [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+            td
+              - if case_details.partial_case_letter_sent_dated.present?
+                time datetime="#{case_details.partial_case_letter_sent_dated}"
+                  = l case_details.partial_case_letter_sent_dated, format: :default
+            td
+              - if case_details.further_actions_required.present?
+                = case_details.further_actions_required.humanize
+
+.partial-flags-buttons
```
Comment (digitalali-moj) [inline app/views/cases/offender_sar/_partial_case_flags.html.slim:27]: Can we use .button-holder instead (if we need it for margin, if not then remove element)

File: config/locales/event.en.yml [hunk 6]
```diff
@@ -66,3 +66,8 @@ en:
     capture_reason_for_lateness: Record reason for lateness
     capture_reason_for_lateness_add: Reason for lateness added
     capture_reason_for_lateness_change: Reason for lateness changed
+    mark_as_partial_case: Case marked as SSCL partial case
+    unmark_as_partial_case: Case marked as not a SSCL partial case
```
Comment (digitalali-moj) [inline config/locales/event.en.yml:70]: Suggesting...Case marked as not **an** SSCL partial case

File: config/locales/event.en.yml [hunk 7]
```diff
@@ -66,3 +66,8 @@ en:
...
+    mark_as_further_actions_required: Case marked as SSCL managing case
+    unmark_as_further_actions_required: Case marked as SSCL not managing case
+    mark_as_awaiting_response_for_partial_case: Case marked as awaiting response following SSCL letter
```
Comment (digitalali-moj) [inline config/locales/event.en.yml:73]: Do we need to say "Awaiting" with upper case A? Just a question as to how the states are written when in sentence.

**Review by digitalali-moj — APPROVED** [review]
Excellent work ⚡  and thanks for fixing my bugs too.
Few minor insignificant non-blocking changes if you feel appropriate.
Great stuff 🎉

**Review by willmcb — APPROVED** [review]
LGTM great job 👍

File: app/views/cases/offender_sar/_partial_case_flags.html.slim (hunk 1 above)
Comment (ymao2) [reply app/views/cases/offender_sar/_partial_case_flags.html.slim]: done

File: app/views/cases/offender_sar/_partial_case_flags.html.slim (hunk 2 above)
Comment (ymao2) [reply app/views/cases/offender_sar/_partial_case_flags.html.slim]: done removed

File: app/models/case/sar/offender.rb (hunk 3 above)
Comment (ymao2) [reply app/models/case/sar/offender.rb:95]: This one don't need, the UI component will do the job,  if somehow the wording is different from the current one in the future which I really doubt, we can put them into translate later without touch this part.

File: app/views/cases/offender_sar/_partial_case_flags.html.slim (hunk 4 above)
Comment (ymao2) [reply app/views/cases/offender_sar/_partial_case_flags.html.slim]: done

File: app/views/cases/offender_sar/_partial_case_flags.html.slim (hunk 5 above)
Comment (ymao2) [reply app/views/cases/offender_sar/_partial_case_flags.html.slim]: done

File: config/locales/event.en.yml (hunk 7 above)
Comment (ymao2) [reply config/locales/event.en.yml:73]: hmm,  I don't know it, I just made a copy of Ryan and NIcols design,  in order to save time to discuss it again,  I just leave it as it is, hopefully you don't mind 😳

**Pull request merged**